src/NexusMods.App.UI/Pages/Library/LibraryView.axaml.cs

```diff
@@ -0,0 +1,61 @@
+using System.Reactive.Disposables;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.ReactiveUI;
+using JetBrains.Annotations;
+using NexusMods.App.UI.Controls;
+using R3;
+using ReactiveUI;
+
+namespace NexusMods.App.UI.Pages.Library;
+
+[UsedImplicitly]
+public partial class LibraryView : ReactiveUserControl<ILibraryViewModel>
+{
+    public LibraryView()
+    {
+        InitializeComponent();
+
+        TreeDataGrid.ElementFactory = new CustomElementFactory();
+
+        this.WhenActivated(disposables =>
+        {
+            Observable.FromEventHandler<TreeDataGridRowEventArgs>(
+                addHandler: handler => TreeDataGrid.RowPrepared += handler,
+                removeHandler: handler => TreeDataGrid.RowPrepared -= handler
+            )
+                .Select(tuple =>
+                {
+                    var (_, args) = tuple;
+                    var row = args.Row;
+
+                    var model = row.Model;
+                    if (model is not Node node) return null;
+                    node.Activate();
+
+                    var cts = new CancellationTokenSource();
+
+                    // NOTE(erri120): this assumes that the row will not be assigned
+                    // a different model between now and when it gets detached
+                    return Observable.FromEventHandler<VisualTreeAttachmentEventArgs>(
+                        addHandler: handler => row.DetachedFromVisualTree += handler,
+                        removeHandler: handler => row.DetachedFromVisualTree -= handler,
+                        cancellationToken: cts.Token
+                    ).Select((node, cts), static (_, tuple) => tuple);
```

Comment:
Could this be not be done with `RowClearing` at top level?
It would just copy the `RowPrepared` but call `node.Deactivate()`

Reply:
Yeah, I'll get that in, thanks